File: util/PNGtoJET.java
package util;

import java.io.*;
import java.util.*;

import com.pdfjet.*;


public class PNGtoJET {

    public static void main(String[] args) throws Exception {
        FileInputStream fis = new FileInputStream(args[0]);
        PNGImage png = new PNGImage(fis);
        byte[] image = png.getData();
        byte[] alpha = png.getAlpha();
        int w = png.getWidth();
        int h = png.getHeight();
        int c = png.getColorType();
        fis.close();

        String fileName = args[0].substring(0, args[0].lastIndexOf("."));
        FileOutputStream fos = new FileOutputStream(fileName + ".jet");
        BufferedOutputStream bos = new BufferedOutputStream(fos);
        writeInt(w, bos);   // Width
        writeInt(h, bos);   // Height
        bos.write(c);       // Color Space
        if (alpha != null) {
            bos.write(1);
            writeInt(alpha.length, bos);
            bos.write(alpha);
        }
        else {
            bos.write(0);
        }
        writeInt(image.length, bos);
        bos.write(image);
        bos.flush();
        bos.close();
    }


    private static void writeInt(int i, OutputStream os) throws IOException {
        os.write((i >> 24) & 0xff);
        os.write((i >> 16) & 0xff);
        os.write((i >>  8) & 0xff);
        os.write((i >>  0) & 0xff);
    }

}   // End of PNGtoJET.java

File: src/main/java/util/PNGtoJET.cs
using System;
using System.IO;
using System.Text;
using System.Collections.Generic;
using System.IO.Compression;

using PDFjet.NET;


namespace util {
public class PNGtoJET {

    public static void Main(String[] args) {
        FileStream fis = new FileStream(args[0], FileMode.Open, FileAccess.Read);
        PNGImage png = new PNGImage(fis);
        byte[] image = png.GetData();
        byte[] alpha = png.GetAlpha();
        int w = png.GetWidth();
        int h = png.GetHeight();
        int c = png.GetColorType();
        fis.Dispose();

        String fileName = args[0].Substring(0, args[0].LastIndexOf("."));
        FileStream fos = new FileStream(fileName + ".jet", FileMode.Create);
        BufferedStream bos = new BufferedStream(fos);
        WriteInt(w, bos);           // Width
        WriteInt(h, bos);           // Height
        bos.WriteByte((byte) c);    // Color Space
        if (alpha != null) {
            bos.WriteByte((byte) 1);
            WriteInt(alpha.Length, bos);
            bos.Write(alpha, 0, alpha.Length);
        }
        else {
            bos.WriteByte((byte) 0);
        }
        WriteInt(image.Length, bos);
        bos.Write(image, 0, image.Length);
        bos.Flush();
        bos.Dispose();
    }


    private static void WriteInt(int i, Stream os) {
        os.WriteByte((byte) (i >> 24));
        os.WriteByte((byte) (i >> 16));
        os.WriteByte((byte) (i >>  8));
        os.WriteByte((byte) (i >>  0));
    }

}   // End of PNGtoJET.cs
}   // End of namespace util
